pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -903,6 +904,11 @@ public boolean isEncryptionRequired() {
         return CompletableFuture.completedFuture(null);
     }
 
+    @Override
+    public Position getLastMessageId() {
+        throw new UnsupportedOperationException("getLastMessageId is not supported on non-persistent topic");
```

Comment:
This should be noted in the javadoc for the client api.

Reply:
Thanks. will change it.